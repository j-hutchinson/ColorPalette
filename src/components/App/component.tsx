import React from 'react';
import styled from 'styled-components';
import Experience from '../Experience/component';
import Introduction from '../Introduction/component'
import ContactMe from '../ContactMe/component';

import { ThemeProvider } from "styled-components";
import { GlobalStyles } from "../../css/GlobalStyle";
import { lightTheme, darkTheme } from "../../css/tokens";
import { useDarkMode } from "../ThemeToggle/useDarkMode";
import Toggle from "../ThemeToggle/component";

const StyledBody = styled.main`
  padding-bottom: 64px;
`;

const App = (): JSX.Element => {
  const [theme, themeToggler] = useDarkMode();
  const themeMode = theme === 'light' ? lightTheme : darkTheme;

  return (
    <ThemeProvider theme={themeMode}>
      <>
        <GlobalStyles />
        <>
          <Toggle toggleTheme={themeToggler} />
          <Introduction />
          <StyledBody >
            <Experience />
            <ContactMe />
          </StyledBody>
        </>
      </>
    </ThemeProvider>
  )
};

export default App;
