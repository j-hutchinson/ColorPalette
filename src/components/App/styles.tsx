import styled from 'styled-components';

export const StyledAppWrapper = styled.div`
    max-width: 1360px;
    margin: auto;
    position: relative;
`;

export const StyledContainer = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-template-rows: 1fr 1fr 1fr;
    gap: 24px;
    width: 100%;
    margin-top: 48px;
`;

export const StyledItem = styled.div`
    background: ${(props: { hex: string }) => props.hex};
    border-radius: 20px;
    box-shadow: 0 0 14px ${(props: { hex: string }) => props.hex};
    display: flex;
    flex-direction: column;
    height: 200px;

    :hover {
        box-shadow: 0 0 30px ${(props: { hex: string }) => props.hex};
    }
`;

export const StyledText = styled.div`
    margin: auto;
    text-align: center;
`;

export const StyledHeader = styled.h2`
    font-size: 20px;
    margin: 8px auto;
`;

export const StyledLabel = styled.p`
    color: black;
    margin: auto;
    font-size: 16px;
    text-transform: uppercase;
`;