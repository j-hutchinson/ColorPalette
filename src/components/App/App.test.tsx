import { shallow } from 'enzyme';
import App from './component';

describe('App component', () => {
    test('component matches snapshot', () => {
        expect.assertions(1);

        const wrapper = shallow(<App />);

        expect(wrapper).toMatchSnapshot();
    });
});