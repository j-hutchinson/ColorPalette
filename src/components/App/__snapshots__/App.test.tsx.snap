// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component component matches snapshot 1`] = `
<Component
  theme={
    Object {
      "background": "linear-gradient(to bottom right, #EDEBEF, #A6A7A4)",
      "border": "#32342B",
      "fontColor": "black",
      "primaryBackground": "#eee",
      "secondaryBackground": "radial-gradient(118.99% 670.46% at -7.06% -9.5%,#FFFFFF 0%,#F1F1F1 100%)",
      "sectionBackground": "radial-gradient(118.99% 670.46% at -7.06% -9.5%,#FFFFFF 0%,#F1F1F1 100%)",
    }
  }
>
  <Memo(l) />
  <Toggle
    toggleTheme={[Function]}
  />
  <Introduction />
  <styled.main>
    <Experience />
    <ContactMe />
  </styled.main>
</Component>
`;
